src/text-buffer.coffee

```diff
@@ -628,6 +645,10 @@ class TextBuffer
     endRow = oldRange.end.row
     rowCount = endRow - startRow + 1
 
+    changeEvent = {oldRange, newRange, oldText, newText}
```

Comment:
You mean using the same object for both events? I'd say each should get their own event object in case the listener modifies it.